benchmarks/microbenchmarks/benchmark_training.py

```diff
@@ -0,0 +1,489 @@
+            #         )
+            #         if hasattr(mock_from_float.return_value, "extra_repr"):
+            #             result.scaling_repr = mock_from_float.return_value.extra_repr()
+            #             print(f"Set scaling_repr to: {result.scaling_repr}")
+
+            # Set fast accumulation if requested
+            if hasattr(float8_model, "forward_config"):
+                float8_model.forward_config = ScaledMMConfig(
+                    False, config.use_fast_accum, False
+                )
+
+            # Store scaling representation for reporting
+            if hasattr(float8_model, "extra_repr"):
+                result.scaling_repr = float8_model.extra_repr()
+
+            # Ensure scaling_repr is not empty
+            if not result.scaling_repr:
+                result.scaling_repr = f"float8 ({config.scaling_granularity})"
+
+            # Define benchmark function for float8 model
+            def float8_forw_backward():
+                float8_model(input_data).sum().backward()
+                float8_model.zero_grad()
+
+            # Wrap the function to execute it multiple times
+            float8_forw_backward_repeated = n_times(REPEAT_N, float8_forw_backward)
+
+            # Compile if requested
+            if config.use_torch_compile:
+                # Inductor settings
+                torch._dynamo.config.cache_size_limit = 1000
+                torch._dynamo.config.automatic_dynamic_shapes = False
+                # torch._dynamo.config.recompile_limit = 10000
+                # torch._dynamo.config.accumulated_recompile_limit = 10000
+
+                print("Compiling Float8 model...")
+                float8_forw_backward_compiled = torch.compile(
+                    float8_forw_backward_repeated
+                )
+
+                # Warmup
+                for _ in range(5):
+                    float8_forw_backward_compiled()
+
+                # Benchmark
+                float8_time = (
+                    benchmark_torch_function_in_microseconds(
+                        float8_forw_backward_compiled
+                    )
+                    * 1e-6
+                    / REPEAT_N
+                )
+            else:
+                # Warmup without compilation
+                for _ in range(5):
+                    float8_forw_backward()
+
+                # Benchmark
+                float8_time = (
+                    benchmark_torch_function_in_microseconds(
+                        float8_forw_backward_repeated
+                    )
+                    * 1e-6
+                    / REPEAT_N
+                )
+
+            # Store float8 time in milliseconds
+            total_time = float8_time * 1000  # Convert to ms
+
+            # For simplicity, we'll estimate forward and backward times
+            # Typically backward is ~2x the forward time for linear layers
+            forward_time = total_time / 3
+            backward_time = total_time * 2 / 3
```

Comment:
I don't understand whats going on here, are those seperate pieces not being calculated seperately?